day1.scala

```diff
@@ -0,0 +1,26 @@
+// using scala 3.0.2
+
+package day1
+
+import scala.io.Source
+
+@main def part1(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): Int = 
+  val depths = input.split('\n').iterator.map(_.toInt)
+  val pairs = depths.sliding(2).map(arr => (arr(0), arr(1)))
```

Comment:
THis could be equivalent to `depths zip depths.tail`

Reply:
oh yes, I don't think I will show it though

Reply:
a prefer sliding.. and it's an intersting method to talk about in the teaching part.

Reply:
The benefits of `zip` is that the types guide you. It gives a collection of pairs that all have exactly two elements. You don’t have to use “unsafe” operations to access the elements of the pair.